iroh-sync/src/ranger.rs

```diff
@@ -412,23 +369,25 @@ where
                             }
                             Err(err) => Some(Err(err)),
                         })
+                        .collect::<Result<Vec<_>, _>>()?;
+                    // add the content status in a second pass
```

Comment:
This adds a new allocation over all items. I guess the borrow checker complains otherwise?
Fine to leave as-is, this code paths will have to optimized further anyway. And #2141 will also touch it.

Reply:
yeah, this is to fight the borrow checker. But allocations are not that expensive compared to db ops.